src/components/icon/SocialIcon.component.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+import FacebookIconSVG from '@/assets/icons/socials/facebook.svg?react';
+import InstagramIconSVG from '@/assets/icons/socials/instagram.svg?react';
+import LinkedinIconSVG from '@/assets/icons/socials/linkedin.svg?react';
+
+export enum SocialIconName {
+    FACEBOOK = 'facebook',
+    INSTAGRAM = 'instagram',
+    LINKEDIN = 'linkedin',
+}
+
+interface SocialIconProps {
+    iconName: SocialIconName;
+    url: string;
+    title?: string;
+}
+
+function getIcon(iconName: SocialIconName, title?: string) {
```

Comment:
So far so good 👍
As an optional enhancement: every switch-case, which you use to render something, could be replaced by the "Map", something like this
```tsx
const SocialIcon: Record<SocialIconName, React.Element> =   {
  facebook: <FacebookIconSVG title={iconTitle} />,
  ...
}


<a href={url} target="_blank" rel="noreferrer">
        {SocialIcon[iconName]}
    </a>

```

Reply:
fixed